sc-tools/sc-builder/src/scg_to_scs_el.cpp

```diff
@@ -0,0 +1,247 @@
+
+const std::unordered_map<std::string, std::string> ScgToScsElement::m_backwardEdgeTypes = {
+    {"pair/const/synonym", "pair/const/-/perm/noorien"},
+    {"pair/const/orient", "pair/const/-/perm/orient"},
+    {"arc/const/fuz", "pair/const/fuz/perm/orient/membership"},
+    {"arc/const/fuz/temp", "pair/const/fuz/temp/orient/membership"},
+    {"arc/const/neg", "pair/const/neg/perm/orient/membership"},
+    {"arc/const/neg/temp", "pair/const/neg/temp/orient/membership"},
+    {"arc/const/pos", "pair/const/pos/perm/orient/membership"},
+    {"arc/const/pos/temp", "pair/const/pos/temp/orient/membership"},
+    {"pair/var/noorient", "pair/-/-/-/noorient"},
+    {"pair/var/orient", "pair/-/-/-/orient"},
+    {"arc/var/fuz", "pair/var/fuz/perm/orient/membership"},
+    {"arc/var/fuz/temp", "pair/var/fuz/temp/orient/membership"},
+    {"arc/var/neg", "pair/var/neg/perm/orient/membership"},
+    {"arc/var/neg/temp", "pair/var/neg/temp/orient/membership"},
+    {"arc/var/pos", "pair/var/-/perm/noorien"},
+    {"arc/var/pos/temp", "pair/var/pos/temp/orient/membership"},
+    {"pair/noorient", "pair/-/-/-/noorient"},
+    {"pair/orient", "pair/-/-/-/orient"},
+    {"arc/-/-", "pair/-/-/-/orient"}};
+
+const std::unordered_map<std::string, std::string> ScgToScsElement::m_unsupportedEdgeTypes = {
+    {"pair/var/-/temp/noorien", "sc_pair_var_temp_noorient"},
+    {"pair/var/-/temp/orient", "sc_pair_var_temp_orient"},
+
+    {"pair/meta/-/perm/noorien", "sc_pair_meta_perm_noorient"},
+    {"pair/meta/-/perm/orient", "sc_pair_meta_perm_orient"},
+    {"pair/meta/-/temp/noorien", "sc_pair_meta_temp_noorient"},
+    {"pair/meta/-/temp/orient", "sc_pair_meta_temp_orient"},
+
+    {"pair/meta/fuz/perm/orient/membership", "sc_pair_meta_fuz_perm_orient_membership"},
+    {"pair/meta/fuz/temp/orient/membership", "sc_pair_meta_fuz_temp_orient_membership"},
+    {"pair/meta/neg/perm/orient/membership", "sc_pair_meta_neg_perm_orient_membership"},
+    {"pair/meta/neg/temp/orient/membership", "sc_pair_meta_neg_temp_orient_membership"},
+    {"pair/meta/pos/perm/orient/membership", "sc_pair_meta_pos_perm_orient_membership"},
+    {"pair/meta/pos/temp/orient/membership", "sc_pair_meta_pos_temp_orient_membership"}};
+
+std::string ScgToScsElement::m_FindValue(
+    std::unordered_map<std::string, std::string> const & dictionary,
+    std::string const & key)
+{
+  auto it = dictionary.find(key);
+  return (it != dictionary.end()) ? it->second : "";
+}
+
+std::string ScgToScsElement::GetElement(std::string const & scgElement, std::string const & dict)
+{
+  static const std::unordered_map<std::string, std::unordered_map<std::string, std::string> const *> dictMap = {
+      {"NodeTypeSets", &m_nodeTypeSets},
+      {"BackwardNodeTypes", &m_backwardNodeTypes},
+      {"UnsupportedNodeTypeSets", &m_unsupportedNodeTypeSets},
+      {"EdgeTypes", &m_edgeTypes},
+      {"BackwardEdgeTypes", &m_backwardEdgeTypes},
+      {"UnsupportedEdgeTypes", &m_unsupportedEdgeTypes}};
```

Comment:
to reduce possibility of error I'd create methods like getNodeType(), getBackwardNodeType() etc.
in that case user won't be using string literals to get required dictionary